cmd/driver/driver_linux.go

```diff
@@ -115,44 +120,60 @@ func NewDriverCmd(ctx context.Context, opt *options.Common) *cobra.Command {
 				// should never happen
 				return fmt.Errorf("unable to retrieve flag type")
 			} else if !f.Changed && viper.IsSet(config.DriverTypeKey) {
-				val := viper.Get(config.DriverTypeKey)
-				if err := cmd.Flags().Set(f.Name, fmt.Sprintf("%v", val)); err != nil {
+				val, err := config.DriverTypes()
+				if err != nil {
+					return err
+				}
+				if err := cmd.Flags().Set(f.Name, strings.Join(val, ",")); err != nil {
 					return fmt.Errorf("unable to overwrite \"type\" flag: %w", err)
 				}
 			}
 
-			if driverTypes.String() != drivertype.TypeAuto {
-				var err error
+			// Logic to discover correct driver to be used
+			// Step 1: build up allowed driver types
+			allowedDriverTypes := make([]drivertype.DriverType, 0)
+			for _, dTypeStr := range driverTypesStr {
 				// Ok driver type was enforced by the user
-				driver.Type, err = drivertype.Parse(driverTypes.String())
+				drvType, err := drivertype.Parse(dTypeStr)
 				if err != nil {
 					return err
 				}
-			} else {
-				// automatic logic
-				info, err := driverkernel.FetchInfo("", "")
-				if err != nil {
-					return err
-				}
-				opt.Printer.Logger.Debug("Fetched kernel info", opt.Printer.Logger.Args(
-					"arch", info.Architecture.ToNonDeb(),
-					"kernel release", info.String(),
-					"kernel version", info.KernelVersion))
-
-				d, err := driverdistro.Discover(info, driver.HostRoot)
-				if err != nil {
-					if !errors.Is(err, driverdistro.ErrUnsupported) {
-						return err
-					}
-					opt.Printer.Logger.Info("Detected an unsupported target system; falling back at generic logic.")
-				}
-				opt.Printer.Logger.Debug("Discovered distro", opt.Printer.Logger.Args("target", d))
+				allowedDriverTypes = append(allowedDriverTypes, drvType)
+			}
 
-				driver.Type = d.PreferredDriver(info)
-				if driver.Type == nil {
-					return fmt.Errorf("automatic driver selection failed")
+			// Step 2: fetch system info (kernel release/version and distro)
+			var err error
+			driver.Kr, err = driverkernel.FetchInfo(driverKernelRelease, driverKernelVersion)
```

Comment:
`options.Driver` now holds run kernelrelease and distro too.
This way, we discover them just once, here, and then pass the driver options down to all driver related commands, without needing to discover distro or kernel infos twice.